cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/handler/sync/RightsizeRequest.java

```diff
@@ -0,0 +1,84 @@
+import com.linkedin.kafka.cruisecontrol.analyzer.ProvisionStatus;
+import com.linkedin.kafka.cruisecontrol.common.Utils;
+import com.linkedin.kafka.cruisecontrol.config.KafkaCruiseControlConfig;
+import com.linkedin.kafka.cruisecontrol.config.constants.AnomalyDetectorConfig;
+import com.linkedin.kafka.cruisecontrol.detector.Provisioner;
+import com.linkedin.kafka.cruisecontrol.detector.ProvisionerState;
+import com.linkedin.kafka.cruisecontrol.detector.RightsizeOptions;
+import com.linkedin.kafka.cruisecontrol.servlet.parameters.RightsizeParameters;
+import com.linkedin.kafka.cruisecontrol.servlet.response.RightsizeResult;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Supplier;
+
+import static com.linkedin.cruisecontrol.common.utils.Utils.validateNotNull;
+import static com.linkedin.kafka.cruisecontrol.detector.AnomalyDetectorUtils.KAFKA_CRUISE_CONTROL_OBJECT_CONFIG;
+import static com.linkedin.kafka.cruisecontrol.servlet.parameters.ParameterUtils.RIGHTSIZE_PARAMETER_OBJECT_CONFIG;
+
+
+public class RightsizeRequest extends AbstractSyncRequest {
+  private static final String RECOMMENDER_UP = "Recommender-Under-Provisioned";
+  protected KafkaCruiseControl _kafkaCruiseControl;
+  protected RightsizeParameters _parameters;
+
+  public RightsizeRequest() {
+    super();
+  }
+
+  @Override
+  protected RightsizeResult handle() {
+    KafkaCruiseControlConfig config = _kafkaCruiseControl.config();
+    Map<String, Object> overrideConfigs;
+    overrideConfigs = Collections.singletonMap(KAFKA_CRUISE_CONTROL_OBJECT_CONFIG, _kafkaCruiseControl);
+    Provisioner provisioner = config.getConfiguredInstance(AnomalyDetectorConfig.PROVISIONER_CLASS_CONFIG,
+                                                           Provisioner.class,
+                                                           overrideConfigs);
+    Supplier<Set<String>> topicNameSupplier = () -> _kafkaCruiseControl.kafkaCluster().topics();
+    Set<String> topicNamesMatchedWithPattern = Utils.getTopicNamesMatchedWithPattern(_parameters.topic(), topicNameSupplier);
+    String topicName;
+    if (topicNamesMatchedWithPattern.size() > 1) {
+      throw new IllegalArgumentException(String.format("The RightsizeEndpoint does not support provisioning for multiple topics {%s}.",
+                                                       String.join(" ,", topicNamesMatchedWithPattern)));
+    } else {
+      topicName = topicNamesMatchedWithPattern.iterator().next();
+    }
+    ProvisionRecommendation recommendation =
+        new ProvisionRecommendation.Builder(ProvisionStatus.UNDER_PROVISIONED).numBrokers(_parameters.numBrokersToAdd())
+                                                                              .numPartitions(_parameters.partitionCount())
+                                                                              .topic(topicName)
+                                                                              .build();
```

Comment:
Isn't this `build()` guaranteed to throw an `IllegalArgumentException`? A `ProvisionRecommendation` with multiple resources (i.e. `numBrokers` and `numPartitions`) is not allowed.

Reply:
Modified the recommendation to go throw the new `sanityCheckResponse` method